lib/browser/guest-window-manager.js

```diff
@@ -83,6 +87,40 @@ const mergeBrowserWindowOptions = function (embedder, options) {
   return options
 }
 
+const MULTIPART_CONTENT_TYPE = 'multipart/form-data'
+const URL_ENCODED_CONTENT_TYPE = 'application/x-www-form-urlencoded'
+function makeContentTypeHeader ({ contentType, boundary }) {
+  const header = `content-type: ${contentType};`
+  if (contentType === MULTIPART_CONTENT_TYPE) {
+    return `${header} boundary=${boundary}`
+  }
+  return header
+}
+
+// Figure out appropriate headers for post data.
+const parseContentTypeFormat = function (postData) {
+  if (postData.length) {
+    // For multipart forms, the first element will start with the boundary
+    // notice, which looks something like `------WebKitFormBoundary12345678`
+    // Note, this regex would fail when submitting a urlencoded form with an
+    // input attribute of name="--theKey", but, uhh, don't do that?
```

Comment:
Should this be documented in our documentation?

Reply:
Something along the lines of urlencoded forms cannot contain inputs that start with `--`

Reply:
Hmm. It's only when you're using a form that's `target="_blank"` _and_ when your webPreferences include `nativeWindowOpen: false`. I think I'm more inclined to make it a bug, since it's not something we can't solve, it's just that we haven't; what do you think? If you still think it should go in the docs, where should it go?

Reply:
@loc I think we should document it in the docs/api/structures/post-body.md doc: https://github.com/electron/electron/pull/19703/files#diff-5f5fc877666cac38e61b5292645c18f1R1

Reply:
The reason for documented this is if/when someone runs into this and files an issue saying that its a bug we can point to docs that state it is not supported.

Reply:
Done!